roles/openshift_inventory/tasks/main.yml

```diff
@@ -0,0 +1,25 @@
+---
+- no_log: True  # keep task descriptions legible
+  block:
+
+  - name: Standardize on latest variable names
+    set_fact:
+      # goal is to deprecate deployment_type in favor of openshift_deployment_type.
+      # both will be accepted for now, but code should refer to the new name.
+      # TODO: once this is well-documented, add deprecation notice if using old name.
+      deployment_type: "{{ openshift_deployment_type | default(deployment_type) | default(None) }}"
+      openshift_deployment_type: "{{ openshift_deployment_type | default(deployment_type) | default(None) }}"
+
+  - name: Normalize openshift_release
+    set_fact:
+      # normalize release if provided, e.g. "v3.5" => "3.5"
+      openshift_release: "{{ openshift_release | string | regex_replace('^v', '') }}"
+    when: openshift_release is defined
```

Comment:
Need to provide some clear comments here and documentation elsewhere about the expectations for `openshift_release`. The official docs don't link [this explanation](https://github.com/openshift/openshift-ansible/blob/403b5c5545b9b07e7a4416e5a63609512ddaa224/inventory/byo/hosts.ose.example#L28) yet, which is what @dgoodwin considers authoritative.

It's not currently required to be defined for an install (looks like upgrades do require it though). Not-defined essentially gets you "latest" although it's a little more complicated than that in `openshift_version`, and in the case of containerized origin you _do_ need to specify `openshift_release` (so we avoid installing alpha builds by default).

Surprisingly, the main use of `openshift_release` doesn't seem to be determining what version to install, but rather verifying that the version to be installed (or upgraded to) matches it. `openshift_version` can figure out what is available in most cases without it, and does not update it according to its findings. When I discussed this with @dgoodwin I came to the conclusion that other variables are eventually defined that determine the actual installation parameters, and we should preserve `openshift_release` as the user's expression of what they wanted, even if they don't specify it.

But it would certainly be clearer if we just required it to be set. It's not documented in the official docs, and the change would break the usage of existing playbooks, but the fix would be pretty easy and obvious.